tools/generator/program.go

```diff
@@ -26,399 +37,280 @@ import (
 	"path/filepath"
 	"regexp"
 	"strings"
+	"time"
 
-	"github.com/Masterminds/semver"
-)
-
-const (
-	defaultRemoteAzureRestAPISpecsPath = "https://github.com/Azure/azure-rest-api-specs.git"
-	defaultAzureRESTAPIBranch          = "current"
-)
-
-// ExitCode gives a hint to the end user about why the program exited without relying on seeing stderr.
-const (
-	ExitCodeOkay int = iota
-	ExitCodeMissingRequirements
-	ExitCodeFailedToClone
-	ExitCodeFinderFailure
+	"github.com/marstr/collection"
 )
 
 var (
-	remoteAzureRESTAPISpecsPath string
-	localAzureRESTAPISpecsPath  string
-	azureRESTAPIBranch          string
-	outputLocation              string
-	dryRun                      bool
-	help                        bool
-	anyMissing                  bool
-	noClone                     bool
-	verbose                     bool
-	wait                        bool
-	targetFile                  string
-	debugLog                    *log.Logger
-	version                     *semver.Version
+	targetFiles    collection.Enumerable
+	packageVersion string
+	outputBase     string
+	logDirBase     string
+	errLog         *log.Logger
+	statusLog      *log.Logger
+	dryRun         bool
 )
 
-func init() {
-	var err error
-	flag.StringVar(&azureRESTAPIBranch, "branch", defaultAzureRESTAPIBranch, "The branch, tag, or SHA1 identifier in the Azure Rest API Specs repository to use during API generation.")
-	flag.StringVar(&remoteAzureRESTAPISpecsPath, "repo", defaultRemoteAzureRestAPISpecsPath, "The path to the location of the Azure REST API Specs repository that should be used for generation.")
-	flag.StringVar(&outputLocation, "output", getDefaultOutputLocation(), "a directory in which all generated code should be placed.")
-	flag.StringVar(&targetFile, "target", "", "If a target file is provided, generator will only run on this file instead of all swaggers it encounters in the repository.")
-	flag.BoolVar(&dryRun, "preview", false, "Use this flag to print a list of packages that would be generated instead of actually generating the new sdk.")
-	flag.BoolVar(&help, "help", false, "Provide this flag to print usage information instead of running the program.")
-	flag.BoolVar(&noClone, "noClone", false, "Use this flag to prevent cloning a new copy of Azure REST API Specs. The existing enlistment should be used instead.")
-	flag.BoolVar(&verbose, "verbose", false, "Print status messages as processing is done.")
-	flag.BoolVar(&wait, "wait", false, "Use this program to halt execution before the cleanup phase is entered.")
-	useDebug := flag.Bool("debug", false, "Include this flag to print debug messages as the program executes.")
-	rawVersion := flag.String("version", "0.0.0", "The version that should be stamped on the generated code. Common usage will be to report user agent.")
-
-	flag.Parse()
-
-	if help {
-		return
-	}
+// version should be set by the linker when compiling this program by providing the following arguments:
+// -X main.version=<version>
```

Comment:
><version> [](start = 19, length = 9)

Here do you mean the semantic version? or the git commit?

Reply:
I've added more clarifying lines later locally.

Though to answer the immediate question, the git commit :)

Reply:
Remind me plz, where do you use the git commit?